frontends/web/src/routes/account/receive/receive.tsx

```diff
@@ -15,376 +15,309 @@
-    // Find index in list of receive addresses that matches the given script type, or -1 if not found.
-    private getIndexOfMatchingScriptType = (scriptType: accountApi.ScriptType): number => {
-        return this.props.receiveAddresses.findIndex(addrs => addrs.scriptType !== null && scriptType === addrs.scriptType);
-    }
-
-    // All script types that are present in the addresses delivered by the backend. Will be empty for if there are no such addresses, e.g. in Ethereum.
-    private availableScriptTypes: accountApi.ScriptType[] = scriptTypes.filter(sc => this.getIndexOfMatchingScriptType(sc) >= 0);
-    public readonly state: State = {
-        verifying: false,
-        activeIndex: 0,
-        paired: null,
-        addressType: 0,
-    }
+export const Receive: FunctionComponent<Props> = ({
+    accounts,
+    code,
+    devices,
+    deviceIDs,
+}) => {
+    const { t } = useTranslation();
+    const [verifying, setVerifying] = useState<boolean>(false);
+    const [activeIndex, setActiveIndex] = useState<number>(0);
+    // index into `availableScriptTypes`, or 0 if none are available.
+    const [addressType, setAddressType] = useState<number>(0);
+    const [addressDialog, setAddressDialog] = useState<AddressDialog>();
+    const [availableScriptTypes, setAvailableScriptTypes] = useState<accountApi.ScriptType[]>();
+    const [currentAddresses, setCurrentAddresses] = useState<accountApi.IReceiveAddress[]>();
+    const [currentAddressIndex, setCurrentAddressIndex] = useState<number>(0);
 
-    public componentDidMount() {
-        if (this.getDevice() === 'bitbox') {
-            apiGet('devices/' + this.props.deviceIDs[0] + '/has-mobile-channel').then(paired => {
-                this.setState({ paired });
-            });
-        }
-    }
+    const device = deviceIDs.length ? devices[deviceIDs[0]] : undefined;
+    const account = accounts.find(({ code: accountCode }) => accountCode === code);
 
-    public UNSAFE_componentWillMount() {
-        this.registerEvents();
-    }
+    // first array index: address types. second array index: unused addresses of that address type.
+    const receiveAddresses = useLoad(accountApi.getReceiveAddressList(code));
+    const secureOutput = useLoad(accountApi.hasSecureOutput(code));
 
-    public componentWillUnmount() {
-        this.unregisterEvents();
-    }
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (e.keyCode === 27 && !verifying) {
+                route(`/account/${code}`);
+            }
+        }
+        document.addEventListener('keydown', handleKeyDown);
+        return () => document.removeEventListener('keydown', handleKeyDown);
+    }, [code, verifying]);
 
-    private registerEvents = () => {
-        document.addEventListener('keydown', this.handleKeyDown);
-    }
+    // Find index in list of receive addresses that matches the given script type, or -1 if not found.
+    const getIndexOfMatchingScriptType = useCallback((
```

Comment:
i find this pretty complicated, why use `useCallback` when it could just be a regular function?

Reply:
yes you are right, I'll just use a normal function, shouldn't matter performance wise.

Reply:
Now I remember why: "TypeScript told me so" also makes a bit sense because the function is used as `useEffect` dependency. So that means: if the dependency array of `useCallback` does not change, then `useCallback` is still the same function and `useEffect` do not have to execute. 

See what happens without `useCallback` (another remark: it still looks complicated `useCallback` doesn't matter IMO)

![Screen Shot 2022-03-14 at 5 58 50 PM](https://user-images.githubusercontent.com/546900/158223301-f70f68d4-a006-4ba2-bef3-20df4903a017.png)

Reply:
done in https://github.com/digitalbitbox/bitbox-wallet-app/pull/1668/commits/d4a2b379a27b867095f321bc367221020d4d69f3

(not squashed yet)